closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -198,6 +271,99 @@ public static boolean isProviderAllowed(Class<?> providerClazz) {
+     */
+    private static void getProfileID(Properties props) {
+        String potentialProfileID = "RestrictedSecurity." + selectedProfile;
+
+        if (selectedProfile.indexOf(".") != -1) {
+            /* The default profile is used, or the user specified the
+             * full <profile.version>.
+             */
+            if (debug != null) {
+                debug.println("Profile specified using full name (i.e., <profile.version>).");
+            }
+            for (Object keyObject : props.keySet()) {
+                // A matching property is found.
+                if (keyObject instanceof String key) {
+                    if (key.startsWith(potentialProfileID)) {
+                        profileID = potentialProfileID;
+                        return;
+                    }
+                }
+            }
+            printStackTraceAndExit(selectedProfile + " is not present in the java.security file.");
+        } else {
+            /* The user specified the only the <profile> without
+             * indicating the <version> part.
+             */
+            if (debug != null) {
+                debug.println("Profile specified without version (i.e., <profile>).");
+            }
+            String defaultMatch = null;
+            for (Object keyObject : props.keySet()) {
+                // A matching property is found.
+                if (keyObject instanceof String key) {
+                    if (key.startsWith(potentialProfileID)) {
+                        // Find the property for specifying if this is the default profile.
+                        if (key.endsWith(".desc.default")) {
+                            // Check if property is set to true.
+                            if (Boolean.parseBoolean(props.getProperty(key))) {
+                                // Check if multiple defaults exist and act accordingly.
+                                if (defaultMatch == null) {
+                                    defaultMatch = key.split("\\.desc")[0];
+                                } else {
+                                    printStackTraceAndExit("Multiple default profiles for " + selectedProfile);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            if (defaultMatch == null) {
+                printStackTraceAndExit("No default profile was found for " + selectedProfile);
+            } else {
+                profileID = defaultMatch;
+            }
+        }
+    }
+
+    private static void checkIfKnownProfileSupported() {
+        if (profileID.contains("NSS") && !isNSSSupported) {
+            printStackTraceAndExit("NSS profiles are not supported in this platform.");
+        }
+
+        if (profileID.contains("OpenJCEPlus") && !isOpenJCEPlusSupported) {
+            printStackTraceAndExit("OpenJCEPlus profiles are not supported in this platform.");
```

Comment:
"on" instead of "in". Pls check the rest, there are more.

Reply:
Fixed.